collections/codec/time.go

```diff
@@ -0,0 +1,51 @@
+package codec
+
+import (
+	"fmt"
+	"time"
+)
+
```

Comment:
not fully agreeing with the implementation, since it seems we just moved the logic from SDK types to collections.

What that issue wants to achieve is also a new design for a sortable time key, this could be achieved, for example, by creating a key which defines time as a unix timestamp with millisecond precision.